lib/trackler/doc_file.rb

```diff
@@ -0,0 +1,92 @@
+class DocFile
+  DEFAULT_IMAGE_PATH = "/docs/img"
+
+  def self.find(basename:, track_dir:, image_path:)
+    dir = File.join(track_dir, "docs")
+
+    [
+      MarkdownFile.new(basename: basename, docs_dir: dir, image_path: image_path),
+      OrgmodeFile.new(basename: basename, docs_dir: dir, image_path: image_path),
+      NullFile.new(basename: basename, docs_dir: dir, image_path: image_path),
+    ].detect(&:acceptable?)
+  end
+
+  attr_reader :basename, :dir, :image_path
+  def initialize(basename:, docs_dir:, image_path:)
+    @basename = basename
+    @dir = docs_dir
+    @image_path = image_path
+  end
+
+  def to_md
+    transform(body).gsub(img_src, img_dst)
+  end
+
+  def name
+    "%s.%s" % [basename, extension]
+  end
+
+  def extension
+    "md"
+  end
+
+  def acceptable?
+    File.exist?(path)
+  end
+
+  private
+
+  def body
+    File.read(path)
+  end
+
+  def path
+    "%s/%s" % [dir, name]
+  end
+
+  def img_src
+    Regexp.new("]\\(%s" % DEFAULT_IMAGE_PATH)
+  end
+
+  def img_dst
+    "](%s" % image_path.gsub(Regexp.new("/$"), "")
+  end
+
+  def transform(s)
+    s
+  end
+end
+
+class OrgmodeFile < DocFile
+  def transform(s)
+    Orgmode::Parser.new(s).to_markdown
+  end
+
+  def extension
+    "org"
+  end
+end
+
+class MarkdownFile < DocFile
+end
+
+class NullFile
+  def initialize(basename:, docs_dir:, image_path:)
+  end
+
+  def to_md
+    ""
+  end
+
+  def name
+    ""
+  end
+
+  def extension
+    "md" # default for missing files
```

Comment:
Are you saying that you feel that this line has risk? Or that the code should be tested whether or not it has risk?